server/chat-plugins/othermetas.ts

```diff
@@ -187,91 +187,113 @@ export const commands: ChatCommands = {
 		const targetid = toID(sep[0]);
 		if (!targetid) return this.parse('/help stone');
 		const stone = getMegaStone(targetid, sep[1]);
-		if (!stone || (dex.gen >= 8 && ['redorb', 'blueorb'].includes(stone.id))) {
-			return this.errorReply(`Error: Mega Stone not found.`);
-		}
+		if (stone && dex.gen >= 8 && ['redorb', 'blueorb'].includes(stone.id)) {
+			return this.errorReply("The Orbs do not exist in Gen 8 and later.");
+		}
+		let stones;
+		if (!stone) {
+			const species = dex.getSpecies(targetid.replace(/(?:mega[xy]?|primal)$/, ''));
+			if (!species.exists) return this.errorReply(`Error: Mega Stone not found.`);
+			if (!species.otherFormes) return this.errorReply(`Error: Mega Evolution not found.`);
+			const megas = species.otherFormes.filter(poke => /(?:-Primal|-Mega(?:-[XY])?)$/.test(poke));
+			if (!megas.length) return this.errorReply(`Error: Mega Evolution not found.`);
+			stones = megas.map(poke => {
+				const megaPoke = dex.getSpecies(poke);
+				const flag = megaPoke.requiredMove === 'Dragon Ascent' ? megaPoke.requiredMove : megaPoke.requiredItem;
+				if (/mega[xy]$/.test(targetid) && toID(megaPoke.name) !== toID(dex.getSpecies(targetid))) return null;
+				if (!flag) return null;
+				return getMegaStone(flag, sep[1]);
+			}).filter(poke => poke !== null);
```

Comment:
I don't think that looping over it twice (in `.map` and `.filter`) is the best plan — a for loop would be just as readable IMO.